sphinx/ext/napoleon/docstring.py

```diff
@@ -1075,14 +1076,60 @@ def __init__(self, docstring: Union[str, List[str]], config: SphinxConfig = None
         super().__init__(docstring, config, app, what, name, obj, options)
 
     def _get_location(self) -> str:
-        filepath = inspect.getfile(self._obj) if self._obj is not None else ""
+        filepath = inspect.getfile(self._obj) if self._obj is not None else None
         name = self._name
 
         if filepath is None and name is None:
             return None
+        elif filepath is None:
+            filepath = ""
 
         return ":".join([filepath, "docstring of %s" % name])
 
+    def _escape_args_and_kwargs(self, name: str) -> str:
+        func = super()._escape_args_and_kwargs
+
+        if ", " in name:
+            args, kwargs, *rest = name.split(", ")
+
+            def check_args(s):
+                return s[:1] == "*" and len([c for c in s if c == "*"]) == 1
+
+            def check_kwargs(s):
+                return s[:2] == "**" and len([c for c in s if c == "*"]) == 2
+
+            is_args = check_args(args)
+            is_kwargs = check_kwargs(kwargs)
+
+            location = self._get_location()
+            if (not is_args and check_args(kwargs)) and (not is_kwargs and check_kwargs(args)):
+                logger.warning(
+                    __("wrong order of *args and **kwargs: %s"),
+                    name,
+                    location=location,
+                )
+            elif (is_args or is_kwargs) and not (is_args and is_kwargs):
+                name_ = args if is_args else kwargs
+                other = "*args" if not is_args else "**kwargs"
+                logger.warning(
+                    __("can only combine parameters of form %s with %s: %s"),
+                    name_,
+                    other,
+                    name,
+                    location=location,
+                )
+            elif is_args and is_kwargs and rest:
+                logger.warning(
+                    __("cannot combine %s and %s with more parameters: %s"),
+                    args,
+                    kwargs,
+                    name,
+                    location=location,
+                )
+            return ", ".join([func(args), func(kwargs)])
```

Comment:
Is this working fine for `x, y, z`?

I feel strange to check syntax in "escaping" function. And it is too much to check the order of arguments. How about doing split-and-escaping only?

Reply:
either is okay to me.

Reply:
then I'll keep it here